sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalSimplification.cs

```diff
@@ -370,13 +360,15 @@ private static bool ExpressionCanBeNull(ExpressionSyntax expression, SemanticMod
 
         private static ImmutableDictionary<string, string> BuildCodeFixProperties(SyntaxNodeAnalysisContext c, string simplifiedOperator = null)
         {
-            var ret = new Dictionary<string, string>();
-            ret.Add(IsCoalesceAssignmentSupportedKey, IsCoalesceAssignmentSupported(c).ToString());
+            var ret = new Dictionary<string, string> {{IsCoalesceAssignmentSupportedKey, IsCoalesceAssignmentSupported(c).ToString()}};
             if (simplifiedOperator != null)
             {
                 ret.Add(SimplifiedOperatorKey, simplifiedOperator);
             }
             return ret.ToImmutableDictionary();
         }
+
+        private static bool IsConditionalStructure(SyntaxNode syntaxNode) =>
```

Comment:
Is this enough? I'd prefer to change the logic to `ContainsConditionalStructure`. This will not work for `return (condition ? a : b)` or some more complex expression (concatenations) containing ternary.

I think we should inspect the nested nodes and stop the nested nodes inspection on invocation syntax. To allow `SomeInvocation(condition ? a : b)`. Those inner arguments should be simple by-design.

Reply:
Like this `syntaxNode.DescendantNodesAndSelf(x => !(x is InvocationExpressionSyntax))`

Reply:
As discussed I will change this to look for ternary op and switch expressions in all descendants (and self).